# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xf86-video-vesa/no-ddc-option.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- xf86-video-vesa-2.0.0/src/vesa.c	2008-07-01 19:40:04.000000000 +0200
+++ xf86-video-vesa-2.0.0-noddc/src/vesa.c	2008-10-20 11:54:13.000000000 +0200
@@ -163,7 +163,8 @@
 typedef enum {
     OPTION_SHADOW_FB,
     OPTION_DFLT_REFRESH,
-    OPTION_MODESET_CLEAR_SCREEN
+    OPTION_MODESET_CLEAR_SCREEN,
+    OPTION_NO_DDC
 } VESAOpts;
 
 static const OptionInfoRec VESAOptions[] = {
@@ -171,6 +172,7 @@
     { OPTION_DFLT_REFRESH, "DefaultRefresh",	OPTV_BOOLEAN,	{0},	FALSE },
     { OPTION_MODESET_CLEAR_SCREEN, "ModeSetClearScreen",
 						OPTV_BOOLEAN,	{0},	FALSE },
+    { OPTION_NO_DDC,	    "NoDDC",		OPTV_BOOLEAN,	{0},	FALSE },
     { -1,		   NULL,		OPTV_NONE,	{0},	FALSE }
 };
 
@@ -625,7 +627,10 @@
 
     xf86SetGamma(pScrn, gzeros);
 
-    if (pVesa->major >= 2) {
+    /* Honour Option "noDDC" */
+    if (pVesa->major >= 2 &&
+	!xf86ReturnOptValBool(pVesa->Options, OPTION_NO_DDC, FALSE))
+    {
 	/* Load ddc module */
 	if ((pDDCModule = xf86LoadSubModule(pScrn, "ddc")) == NULL) {
 	    vbeFree(pVesa->pVbe);
